test/parallel/test-url-parse-invalid-input.js

```diff
@@ -13,8 +13,9 @@ const url = require('url');
   0,
   [],
   {}
-].forEach(function(val) {
-  assert.throws(function() { url.parse(val); }, TypeError);
+].forEach((val) => {
+  assert.throws(() => { url.parse(val); },
+                /^TypeError: Parameter "url" must be a string, not (undefined|boolean|number|object)$/);
 });
 
-assert.throws(function() { url.parse('http://%E0%A4%A@fail'); }, /^URIError: URI malformed$/);
+assert.throws(() => { url.parse('http://%E0%A4%A@fail'); }, /^URIError: URI malformed$/);
```

Comment:
I believe this one is still long. I'm fine with the inline regex being whitelisted, but I think if there are multiple arguments we should still try to split them by line to avoid the long lines if possible.